src/chains.rs

```diff
@@ -26,14 +26,27 @@ use crate::device::Device as DeviceAPI;
 use crate::{ContextID, Error, SurfaceAccess, SurfaceInfo, SurfaceType};
 use euclid::default::Size2D;
 use fnv::{FnvHashMap, FnvHashSet};
+use glow as gl;
+use glow::Context as Gl;
+use glow::HasContext;
 use log::debug;
-use sparkle::gl::{self, GLuint, Gl};
 use std::collections::hash_map::Entry;
 use std::fmt::Debug;
 use std::hash::Hash;
 use std::mem;
+use std::num::NonZero;
 use std::sync::{Arc, Mutex, MutexGuard, RwLock, RwLockReadGuard, RwLockWriteGuard};
 
+fn framebuffer(framebuffer: u32) -> Option<gl::NativeFramebuffer> {
+    NonZero::new(framebuffer).map(gl::NativeFramebuffer)
+}
```

Comment:
Let's move this into the function where it is used now that it is only used in one place.

Reply:
I think it's okay to duplicate this code which is one line.

Reply:
Actually we can use `get_parameter_framebuffer` (currently not yet release in glow) to eliminate last uses of framebuffer function.

Reply:
That sounds perfect.